lib/mongo/auth/aws/credentials_retriever.rb

```diff
@@ -188,6 +194,48 @@ def ecs_metadata_credentials
           return nil
         end
 
+        def assume_role_with_wed_identity_credentials
```

Comment:
I know I'm the new guy, so forgive me if my suggestions are overstepping any bounds, but since I can't really comment (yet) on the functionality of the PR, I figured I might as well share my thoughts on style and architecture. :)

Especially as the new guy, looking at this method is a bit challenging; it requires careful picking apart to see what's going on and what the different pieces are being used for. However, there are independent sections of this method that can be extracted into auxiliary methods, which can make the original method more concise, and clarify its intent.

For example:

```ruby
def assume_role_with_web_identity_credentials
  web_identity_token, role_session_name = prepare_web_identity_inputs
  return if web_identity_token.nil?

  response = request_web_identity_role(web_identity_token, role_session_name)
  return if response.nil?

  credentials_from_web_identity_response(response)
end
```

Reply:
Great idea, thank you! Changed accordingly.